modules/oe_content_sub_entity/modules/oe_content_sub_entity_document_reference/tests/src/Functional/DocumentReferenceBundlesTest.php

```diff
@@ -0,0 +1,234 @@
+    $subform->fillField(
+      'Use existing media',
+      $document->label() . ' (' . $document->id() . ')',
+    );
+    $subform->pressButton('Create document reference');
+
+    // Add a document group.
+    $subform->selectFieldOption('node_docs[actions][bundle]', 'Document group');
+    $subform->pressButton('Add new document reference');
+    $subform->fillField('Title', 'Example documents group');
+    $document = $this->createDocumentMedia(1);
+    $subform->fillField(
+      'node_docs[form][1][oe_documents][0][target_id]',
+      $document->label() . ' (' . $document->id() . ')',
+    );
+    $subform->pressButton('Add another item');
+    $document = $this->createDocumentMedia(2);
+    $subform->fillField(
+      'node_docs[form][1][oe_documents][1][target_id]',
+      $document->label() . ' (' . $document->id() . ')',
+    );
+    $subform->pressButton('Create document reference');
+    $page->pressButton('Save');
+
+    // Login with visitor-like permissions.
+    $this->drupalLogin($this->drupalCreateUser([
+      'access content',
+      'view media',
+    ]));
+    $this->drupalGet('node/1');
+
+    // Check the newly created page.
+    $assert_session->elementTextEquals('css', 'h1.page-title', 'Example page');
+    $assert_session->elementsCount('css', '.media', 3);
+    $assert_session->elementsCount('css', '.file', 3);
+
+    // Get document reference field items.
+    $references = $page->findAll('css', '.field--name-node-docs > .field__items > .field__item');
+
+    // Check single document reference.
+    $assert_session->elementsCount('css', '.media', 1, $references[0]);
+    $assert_session->elementsCount('css', '.file', 1, $references[0]);
+    $link = $references[0]->find('css', '.media .file a');
+    $this->assertSame('text-0.txt', $link->getText());
+    $this->assertStringContainsString('files/text-0', $link->getAttribute('href'));
+
+    // Check the document group.
+    $this->assertSame(
+      'Example documents group',
+      $assert_session
+        ->elementExists('css', '.field--name-oe-title > .field__item', $references[1])
+        ->getText()
+    );
+    $links = $references[1]->findAll('css', '.media .file a');
+    $this->assertSame('text-1.txt', $links[0]->getText());
+    $this->assertStringContainsString('files/text-1', $links[0]->getAttribute('href'));
+    $this->assertSame('text-2.txt', $links[1]->getText());
+    $this->assertStringContainsString('files/text-2', $links[1]->getAttribute('href'));
+  }
+
+  /**
+   * Creates a document media entity.
+   *
+   * @param int $index
+   *   Index to distinguish from other documents.
+   *   The index will be included in all labels, and is also used to pick a
+   *   specific file entity.
+   *   This is limited by the number of files returned from ->getTestFiles().
+   *
+   * @return \Drupal\media\MediaInterface
+   *   The media entity.
+   */
+  protected function createDocumentMedia(int $index): MediaInterface {
```

Comment:
again we have development like this when I think it should go into https://github.com/openeuropa/oe_media/pull/124/files

Reply:
Not needed now.